tests/common/test_event.py

```diff
@@ -453,3 +454,131 @@ def test_report_metric(self, mock_event):
                 break
         else:
             self.fail("Counter '%idle' not found in event parameters: {0}".format(repr(event_dictionary)))
+
+    @patch("azurelinuxagent.common.event.get_container_id_from_env", return_value="TEST_CONTAINER_ID")
+    def test_add_default_parameters_to_extension_event(self, *args):
+        # When no values are populated in the TelemetryEventParamList.
+        default_parameters_expected = {"GAVersion": CURRENT_AGENT,
+                                       'ContainerId': "TEST_CONTAINER_ID",
+                                       'OpcodeName': "",
+                                       'EventTid': 0,
+                                       'EventPid': 0,
+                                       "TaskName": "",
+                                       "KeywordName": ""}
+        default_parameters_expected_names = set(default_parameters_expected.keys())
+        extension_param_list_empty = EventLogger.add_default_parameters_to_event([], set_values_for_agent=False)
+
+        # Converting list of TelemetryEventParam into a dictionary, for easier look up of values.
+        param_list_dict = OrderedDict([(param.name, param.value) for param in extension_param_list_empty])
+
+        counter = 0
+        for p in default_parameters_expected_names:
+            self.assertIn(p, param_list_dict)
+            self.assertEqual(param_list_dict[p], default_parameters_expected[p])
+            counter += 1
+
+        self.assertEqual(len(default_parameters_expected_names), counter)
+
+        # When some values are already populated in the TelemetryEventParamList.
+        extension_param_list_populated = [TelemetryEventParam('Name', "DummyExtension"),
+                                          TelemetryEventParam('Version', CURRENT_VERSION),
+                                          TelemetryEventParam('Operation', "DummyOperation"),
+                                          TelemetryEventParam('OperationSuccess', True),
+                                          TelemetryEventParam('Message', "TestMessage"),
+                                          TelemetryEventParam('Duration', 10), TelemetryEventParam('ExtensionType', ''),
+                                          TelemetryEventParam('OpcodeName', '')]
+
+        extension_param_list_with_defaults = EventLogger.add_default_parameters_to_event(
+            extension_param_list_populated, set_values_for_agent=False)
+
+        # Converting list of TelemetryEventParam into a dictionary, for easier look up of values.
+        param_list_dict = OrderedDict([(param.name, param.value) for param in extension_param_list_with_defaults])
+
+        counter = 0
+        for p in default_parameters_expected_names:
```

Comment:
I see this logic being used in multiple places. Why not make this whole into a helper private function and just call that for better code cleanup and readability?

Reply:
Done.

Reply:
Done.